src/main/java/tds/common/NoContentResponse.java

```diff
@@ -0,0 +1,27 @@
+package tds.common;
+
+import java.util.Optional;
+
+/**
+ * Contains errors (if any) when responding without data (e.g. when the response would be a 204 - No Content).
+ */
+public class NoContentResponse {
+    private ValidationError[] errors;
+
+    /**
+     * @param errors errors present when completing the request
+     */
+    public NoContentResponse(ValidationError... errors) {
+        this.errors = errors;
+    }
+
+    /**
+     * @return errors when completing the request.
+     */
+    public Optional<ValidationError[]> getErrors() {
```

Comment:
Is there a reason this needs to be Optional instead of just returning an empty array?